### DISCUSSION TITLE
Option to not wrap SVG images into <img> tag?

### DISCUSSION THREAD

**Discussion by liquidcarbon** [quarto-dev/quarto-cli#4467] [Q&A]
Hi, I work with a lot of SVG images generated in the outputs of Jupyter notebooks.
Quarto extracts the images from the output cells, saves them as individual SVG files and wraps them into `<img>` tags, which makes them lose native SVG functionality (such as the ability to have clickable hyperlinks in them - like the text in blue here).

Is there a way to keep SVGs as native SVGs?

![image](https://user-images.githubusercontent.com/47034358/220779710-12133252-a1d6-4aab-9297-d48e6923eac6.png)

**Comment by cscheid** [comment]
Can you share a full, minimal example? Thanks!

**Reply by liquidcarbon** [reply]
sure, it's the last image on this page: https://liqc.quarto.pub/a-million-little-monomers/posts/2023-02-04-are-u-rdkitting-me/2023-02-04-rdkitting.html
I would like the SVG to render like this (with hyperlink): https://liqc.quarto.pub/a-million-little-monomers/posts/2023-02-04-are-u-rdkitting-me/2023-02-04-rdkitting_files/figure-html/cell-8-output-1.svg

I think it has to do with SVG being able to break things: https://stackoverflow.com/questions/13808020/include-an-svg-hosted-on-github-in-markdown
If there could be a "use at your own risk" option, that would be really helpful though.

**Reply by cscheid** [reply]
Sorry, I should have clarified that I need the project _input_, not the output.

**Reply by liquidcarbon** [reply]
The input is this notebook.

Github says "don't support that file type" on trying upload ipynb, so I renamed it .ipynb.zip - just remove the `.zip` part, it's not a real zipped notebook.
[2023-02-04-rdkitting.ipynb.zip](https://github.com/quarto-dev/quarto-cli/files/10809648/2023-02-04-rdkitting.ipynb.zip)

**Reply by liquidcarbon** [reply]
but maybe worth mentioning that it does the same thing to SVG files when you call them using Markdown `![]()` syntax.  The also get packed into `<img>`

**Reply by cscheid** [reply]
I see. This is coming from Pandoc, and it's the desired behavior most of the time.

I think the best way to do this is for you to solve it yourself with a Lua filter, where you'd search for the `Image` AST node, find the svg files you're interested in replacing, and then replacing the image with a `RawBlock` node that has the contents you're seeking. I haven't tested this, though, and it might have negative interactions with our crossref system, sadly, if your SVG references are in places where we expect a block with an image.

**Reply by liquidcarbon** [reply]
Uff, that's all new to me.  Do you have some pointers?

**Reply by cscheid** [reply]
I apologize - it's definitely a bit of a learning curve. We have some pointers here: https://quarto.org/docs/extensions/filters.html#overview

Unfortunately, I don't think we want to add this feature to quarto ourselves, because it won't always work (the svg is interpreted in a different CSS context, etc). So the best solution really is a filter in the situations where it's needed, but you're right, it will require learning about Pandoc's filters and Lua.

**Comment by liquidcarbon** [comment]
Found a workaround:

```
class SneakySVG(str):
    """A string that is rendered as HTML code."""
    def show(self):
        from IPython.display import HTML
        return HTML(self)
```

Call like so from a notebook:
```
SneakySVG("""
<svg>

</svg>
""").show()
```

When you render the notebook, you get SVG right in your rendered page, and no separate `_files` folder with output files generated.  Win!
![image](https://user-images.githubusercontent.com/47034358/221338313-5d7b1669-595b-45ae-9a2f-a48fe7c498d0.png)

And it even responds to additional styling you put into root `/styles.css` :D

**Comment by gly-git** [comment]
Hello,
First, thank you for the sharing.
I am facing the same need as liquidcarbon.
I have first followed the recommandation of cscheid and wrote a Lua filter to have "clickable" links in a SVG image. The problem I am facing with this approach is that the SVG image do not respects the specified layout (e.g. {.column-body}) of the page. 
I have the same result with the workaround suggested by liquidcarbon.
Is there a way to apply the specified layout to a raw SVG ?
